book/src/specs/kimchi.md

```diff
@@ -1732,7 +1732,7 @@ pub struct ProverIndex<G: KimchiCurve, OpeningProof: OpenProof<G>> {
     pub max_poly_size: usize,
 
     #[serde(bound = "ColumnEvaluations<G::ScalarField>: Serialize + DeserializeOwned")]
-    pub column_evaluations: ColumnEvaluations<G::ScalarField>,
+    pub column_evaluations: Arc<LazyCache<ColumnEvaluations<G::ScalarField>>>,
```

Comment:
More for a discussion (no changes requested): we're adding many indirection here, even if it seems needed.
I hope it won't affect performances too much. I don't think so as the computation that will happen with the data resulting to accessing the additional indirection are way more expensive than going through memory.

I'm wondering how this will affect o1js - I don't have any idea.

Reply:
Is it three indirections?
Arc -> LazyCache -> Option (for the field value) -> T

`UnsafeCell` is `transparent` (see https://doc.rust-lang.org/src/core/cell.rs.html#2073), so there is no additional pointer.

Reply:
Also, even if it is negligible, we add a (really small) additional memory footprint. Explicitly:
- Some for the additional data in `Arc` (i.e. `strong` and `weak` -> 2 * 64 bits).
- Some for the fields in LazyCache -> `once`, `value` and `init`.

Only a few words, but still interesting to notice.

Reply:
We only have one `ProverIndex` per proof, so it is fine.